File: problems/problems_3304/Solution.java
package problems.problems_3304;

import com.alibaba.fastjson.JSON;
import java.util.*;
import qubhjava.BaseSolution;


public class Solution extends BaseSolution {
    public char kthCharacter(int k) {
        int count = 0;
        while (k > 1) {
            k -= 1 << (31 - Integer.numberOfLeadingZeros(k - 1));
            count++;
        }
        return (char) ('a' + count % 26);
    }

    @Override
    public Object solve(String[] inputJsonValues) {
        int k = Integer.parseInt(inputJsonValues[0]);
        return JSON.toJSON(kthCharacter(k));
    }
}

File: problems/problems_3304/solution.py
import solution
from typing import *


class Solution(solution.Solution):
    def solve(self, test_input=None):
        return self.kthCharacter(test_input)

    def kthCharacter(self, k: int) -> str:
        steps = 0
        while k > 1:
            k -= 1 << ((k - 1).bit_length() - 1)
            steps += 1
        return chr(ord('a') + steps % 26)
